spec/syntax.md

```diff
@@ -148,19 +148,35 @@ MAY include an _annotation_ that is applied to the external value.
 
 A **_<dfn>local-declaration</dfn>_** binds a _variable_ to the resolved value of an _expression_.
 
-Declared _variables_ MUST NOT be used before their _declaration_,
-and their values MUST NOT be self-referential;
-otherwise, a _message_ is not considered _valid_.
-
-Multiple _declarations_ MUST NOT bind a value to the same _variable_;
-otherwise, a _message_ is not considered _valid_.
-
 ```abnf
 declaration = input-declaration / local-declaration
 input-declaration = input [s] variable-expression
 local-declaration = local s variable [s] "=" [s] expression
 ```
 
+_Variables_, once declared, MUST NOT be redeclared. 
+A _message_ that does any of the following is not _valid_ and will produce a 
+Duplicate Declaration error during formatting:
+- An _input-declaration_ MUST NOT bind a _variable_ that appears as a _variable_ in a previous 
+  _declaration_.
+- A _local-declaration_ MUST NOT bind a _variable_ that appears as a _variable_ in a previous
+  _declaration_.
+- A _local-declaration_ MUST NOT bind a _variable_ that appears in its _expression_.
+
+> [!Note]
+> These restrictions only apply to _declarations_.
+> A _placeholder_ or _selector_ MAY override the annotation provided in a _declaration_.
```

Comment:
I think it's very confusing to call this "overriding", since no new name is being introduced. It's just constructing a new expression.

Reply:
Fair enough. Try the wording changes in the e9908c8.